src/Sarif/Visitors/SarifVersionOneToCurrentVisitor.cs

```diff
@@ -243,21 +243,106 @@ public Rule TransformRuleVersionOne(RuleVersionOne node)
             return rule;
         }
 
+        private List<FileLocation> CreateFileLocationsFromDictionary(IDictionary<string, string> dictionary, Run run)
+        {
+            List<FileLocation> result = null;
+
+            if (dictionary != null)
+            {
+                result = new List<FileLocation>();
+
+                foreach (string key in dictionary.Keys)
+                {
+                    var fileLocation = new FileLocation
+                    {
+                        Uri = new Uri(key, UriKind.RelativeOrAbsolute)
+                    };
+                    result.Add(fileLocation);
+
+                    if (run != null && !string.IsNullOrWhiteSpace(dictionary[key]))
+                    {
+                        // We have contents, so mention this file in run.files
+                        if (run.Files == null)
+                        {
+                            run.Files = new Dictionary<string, FileData>();
+                        }
+
+                        if (!run.Files.ContainsKey(key))
+                        {
+                            run.Files.Add(key, new FileData
+                            {
+                                Contents = new FileContent
+                                {
+                                    Text = dictionary[key]
+                                },
+                                FileLocation = fileLocation
+                            });
+                        }
+                    }
+                }
+            }
+
+            return result;
+        }
+
+        public Invocation TransformInvocationVersionOne(InvocationVersionOne node)
+        {
+            Invocation invocation = null;
+
+            if (node != null)
+            {
+                invocation = new Invocation
+                {
+                    Account = node.Account,
+                    CommandLine = node.CommandLine,
+                    EndTime = node.EndTime,
+                    EnvironmentVariables = node.EnvironmentVariables,
+                    Machine = node.Machine,
+                    ProcessId = node.ProcessId,
+                    Properties = node.Properties,
+                    ResponseFiles = CreateFileLocationsFromDictionary(node.ResponseFiles, SarifLog.Runs.Last()),
+                    StartTime = node.StartTime,
+                    WorkingDirectory = node.WorkingDirectory
+                };
+
+                if (!string.IsNullOrWhiteSpace(node.FileName))
+                {
+                    if (!string.IsNullOrWhiteSpace(node.CommandLine))
+                    {
+                        string exeName = Path.GetFileName(node.FileName);
+                        int startIndex = node.CommandLine.IndexOf(exeName) + exeName.Length;
```

Comment:
This is wrong. `fileName` might be `"C:\Program Files (x86)\CodeScanner\scanner.exe"`, but `commandLine` might be "`scanner *.cs`" (if `scanner.exe` were on your path). That's why (as I mentioned above) `fileName` exists at all.